src/openMVG/sfm/pipelines/localization/SfM_Localizer.cpp

```diff
@@ -105,6 +114,81 @@ bool SfM_Localizer::Localize
       // Update the upper bound precision of the model found by AC-RANSAC
       resection_data.error_max = ACRansacOut.first;
     }
+#else
+    
+    // since the intrinsics are known undistort the input 2d points
+    //@fixe there is a lot of code redundancy in this solution; find better solution to
+    // avoid code duplication when calling ACRansacOut
+    Mat pt2Dundistorted;
+    const bool has_disto = pinhole_cam->have_disto();
+    if(has_disto)
+    {
+      const std::size_t numPts = resection_data.pt2D.cols();
+      pt2Dundistorted = Mat2X(2, numPts);
+      for(std::size_t iPoint = 0; iPoint < numPts; ++iPoint)
+      {
+        pt2Dundistorted.col(iPoint) = pinhole_cam->get_ud_pixel(resection_data.pt2D.col(iPoint));
+      }
+    }
+    
+    if(estimator == robust::ROBUST_ESTIMATOR_ACRANSAC)
+    {
+      //--
+      // Since K calibration matrix is known, compute only [R|t]
+      typedef openMVG::euclidean_resection::P3PSolver SolverType;
+      MINIMUM_SAMPLES = SolverType::MINIMUM_SAMPLES;
+
+      typedef openMVG::robust::ACKernelAdaptorResection_K<
+        SolverType, ResectionSquaredResidualError,
+        openMVG::robust::UnnormalizerResection, Mat34>  KernelType;
+
+      // otherwise we just pass the input points
+      KernelType kernel = KernelType(has_disto ? pt2Dundistorted : resection_data.pt2D, resection_data.pt3D, pinhole_cam->K());
+
+      // Robust estimation of the Projection matrix and its precision
+      const std::pair<double, double> ACRansacOut =
+              openMVG::robust::ACRANSAC(kernel, resection_data.vec_inliers, resection_data.max_iteration, &P, dPrecision, true);
+      // Update the upper bound precision of the model found by AC-RANSAC
+      resection_data.error_max = ACRansacOut.first;
+    }
+    else if(estimator == robust::ROBUST_ESTIMATOR_LORANSAC)
+    {
+      
+      // just a safeguard
+      if(resection_data.error_max == std::numeric_limits<double>::infinity())
```

Comment:
Could be nice to have a way to use the "default" value of each *sac algorithm.
It could be infinity or -1 or 0...